### PR TITLE
Expand golden-test docs in README

### PR DESCRIPTION
**Pull request by drevell**
There's nothing really new here, I'm just adding more material and clarifying a few things.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,35 +113,69 @@ The valid values for `ABC_LOG_LEVEL` are `debug`, `info`, `notice`, `warning`,
...
 The `<test_name>` parameter gives the test names to record or verify, if not
 specified, all tests will be run against. This flag may be repeated, like
 `--test-name=test1`, `--test-name=test2`, or `--test-name=test1,test2`.
 
 The `<location>` parameter gives the location of the template.
 
-For every test case, it is expected that
+For every test case, it is expected that a
+`testdata/golden/<test_name>/test.yaml` exists to define template input
+params.Each "input" in this file must correspond to a template input defined in
```
Comment (bradegler) [inline README.md:151]:
```suggestion
params. Each "input" in this file must correspond to a template input defined in
```

File: README.md [hunk 2]
```diff
@@ -113,35 +113,69 @@ The valid values for `ABC_LOG_LEVEL` are `debug`, `info`, `notice`, `warning`,
...
+the template's`spec.yaml`. Each required input in the template's spec.yaml must
```
Comment (bradegler) [inline README.md:152]:
```suggestion
the template's `spec.yaml`. Each required input in the template's spec.yaml must
```

**Review by bradegler — CHANGES_REQUESTED** [review]
Two minor spacing issues, otherwise LGTM.

File: README.md (hunk 1 above)
Comment (drevell) [reply README.md]: 🤦

File: README.md (hunk 2 above)
Comment (drevell) [reply README.md]: 🤦

**Review by bradegler — DISMISSED** [review]

**Review by gjonathanhong — APPROVED** [review]

**Pull request merged**